pkg/resource/deploy/source_eval_test.go

```diff
@@ -221,7 +223,8 @@ func TestRegisterNoDefaultProviders(t *testing.T) {
 	ctx, err := newTestPluginContext(t, fixedProgram(steps))
 	assert.NoError(t, err)
 
-	iter, err := NewEvalSource(ctx, runInfo, nil, false).Iterate(context.Background(), Options{}, &testProviderSource{})
+	evaluationSource := NewEvalSource(ctx, runInfo, nil, false, nil)
```

Comment:
We'll need a dummy noop loader for these tests that don't use schema. I'd rather be strict and assert that things are set and deal with dummy placeholders in tests than being lax about contract assertions.